framework/bundles/org.jboss.tools.rsp.server/src/main/java/org/jboss/tools/rsp/server/jobs/JobManager.java

```diff
@@ -26,34 +38,62 @@ public void removeJobListener(IJobListener l) {
 	}
 
 	@Override
-	public void addJob(IJob job) {
+	public IJob scheduleJob(String jobName, IRunnableWithProgress runnable) {
+		SimpleJob job = new SimpleJob(jobName, generateJobId(), runnable);
 		currentJobs.put(job.getId(), job);
-		ArrayList<IJobListener> tmp = new ArrayList<>(listeners);
-		for( IJobListener l : tmp ) {
-			l.jobAdded(job);
-		}
+		fireJobAdded(job);
+		schedule(job);
+		return job;
 	}
 
 	@Override
-	public void removeJob(IJob job, IStatus status) {
-		currentJobs.remove(job.getId());
+	public IJob scheduleJob(String jobName, IStatusRunnableWithProgress runnable) {
+		SimpleJob job = new SimpleJob(jobName, generateJobId(), runnable);
+		currentJobs.put(job.getId(), job);
+		fireJobAdded(job);
```

Comment:
1) firing is duplicate. I'd personally prefer the firing outside of this method, it would keep this method focused on a single task: scheduling.
2) same question as above: shouldn't firing happen after the scheduling of the job?